mypy/checker.py

```diff
@@ -1916,10 +2045,12 @@ def visit_yield_expr(self, e: YieldExpr) -> Type:
     def check_subtype(self, subtype: Type, supertype: Type, context: Context,
                       msg: str = messages.INCOMPATIBLE_TYPES,
                       subtype_label: str = None,
-                      supertype_label: str = None) -> None:
+                      supertype_label: str = None) -> bool:
```

Comment:
I think it'd be cleaner to use `is_subtype` + directly calling `self.fail` in the one place you use this return value.  This doesn't feel idiomatic with the other uses.